### ISSUE TITLE
Permit "default" as value for "optimizerSteps"

### ISSUE THREAD

**Issue by PaulRBerg** [foundry-rs/foundry#3904]
### Component

Forge

### Describe the feature you would like

The default Yul optimizer step sequence is quite long-winded - even in the Solidity source code, the default value is [split on multiple lines](https://github.com/ethereum/solidity/blob/73fcf69188fed78c3ad91f81ce7d6ed7c6ee79c6/libsolidity/interface/OptimiserSettings.h#L44-L60) for readability purposes.

My feature request is to make it possible to pass the `default` value to the `optimizerSteps` field in the Foundry config, so that users don't have to manually write it up:

```toml
[profile.default.optimizer_details.yulDetails]
stackAllocation = true
optimizerSteps = "default"
```

This is useful because users may want to disable the default step sequence during development (i.e. in the `default` profile) in order to speed up the compilation time. But then, in a separate profile such as `production`, users may want to set the step sequence back to the default to get maximum optimization benefits.

### Additional context

See the conversation [here](https://twitter.com/PaulRBerg/status/1604132363700690944).

**Comment by mattsse** [comment]
supportive of that!

**Comment by mds1** [comment]
One caveat here is that it's possible the default sequence of optimizer steps will change between solc versions, and if so we'd need to stay on top of that with each solc release and maintain a mapping from version to step sequence. (Or, maybe it's unlikely to change / will never change, not sure)

**Comment by paulrberg** [comment]
Yeah, it's possible that the default sequence will change with time. In this case, I suggest maintaining a mapping as you @mds1 suggest, and throwing an error for those versions for which there is no default value available in Foundry.

**Comment by cameel** [comment]
Why not just omit the option from the JSON passed to the compiler (or from its CLI arguments) when it's default? You would not have to keep the mapping. Changes to the sequence aren't rare, especially when we're tweaking the optimizer.

Also, using specifically `""` default to indicate the default does not seem like a good idea because it's in fact a valid sequence: `VarDeclInitializer`, `ExpressionInliner`, `BlockFlattener`, `SSATransform`, `UnusedPruner`, `CircularReferencesPruner`, `StructuralSimplifier`. :)

**Comment by zerosnacks** [comment]
Per comment above marking this as `not planned` given "default" itself would be a valid sequence and it is very rare for users to want to modify the optimizer steps I think and if they do it is likely they will need to maintain the specified sequence manually. Feel free to leave a comment if any of you feel strongly about this and I'll re-open.
